libs/simdvec/src/main21/java/org/elasticsearch/simdvec/internal/vectorization/PanamaESVectorUtilSupport.java

```diff
@@ -1122,4 +1122,27 @@ private void transposeHalfByte128(int[] q, byte[] quantQueryByte) {
         quantQueryByte[index + quantQueryByte.length / 2] = (byte) upperMiddleByte;
         quantQueryByte[index + 3 * quantQueryByte.length / 4] = (byte) upperByte;
     }
+
+    @Override
+    public int indexOf(final byte[] bytes, final int offset, final int length, final byte marker) {
+        final ByteVector markerVector = ByteVector.broadcast(ByteVector.SPECIES_PREFERRED, marker);
+        final int upperBound = offset + ByteVector.SPECIES_PREFERRED.loopBound(length);
+
+        int i = offset;
+        for (; i < upperBound; i += ByteVector.SPECIES_PREFERRED.length()) {
+            ByteVector chunk = ByteVector.fromArray(ByteVector.SPECIES_PREFERRED, bytes, i);
+            VectorMask<Byte> mask = chunk.eq(markerVector);
+            if (mask.anyTrue()) {
+                return (i - offset) + mask.firstTrue();
+            }
+        }
```

Comment:
Should we guard vectorization for when length is smaller than SPECIES_PREFERRED?

It seems at a minimum for each SPECIES_PREFERRED, if `length < ByteVector.SPECIES_PREFERRED.length()` this should pivot back to `ByteArrayUtils.indexOf`

Sometimes, its conceivable that we are searching towards the end and only have 512, 256, or even 128 bytes left, thus causing the slower tail to then dominate the cost, when the `ByteArrayUtils.indexOf` would still unroll, etc.

Reply:
no need.